### ISSUE TITLE
Profile may be null

### ISSUE THREAD

**Issue by trixpan** [keycloak/keycloak#35583]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

authentication

### Describe the bug

As part of https://github.com/keycloak/keycloak/issues/28840 a call to `Profile.class` is made from `FreeMarkerLoginFormsProvider`

```
  if (Profile.isFeatureEnabled(Feature.ORGANIZATION) && realm.isOrganizationsEnabled()) {
                idpBean = new OrganizationAwareIdentityProviderBean(idpBean);
            }
```

This caused a regression when running authenticator / form test units which used to work before 25.x

### Version

26.0.7

### Regression

- [X] The issue is a regression

### Expected behavior

unit tests pass

### Actual behavior

```
java.lang.NullPointerException: Cannot read field "features" because the return value of "org.keycloak.common.Profile.getInstance()" is null

	at org.keycloak.common.Profile.isFeatureEnabled(Profile.java:392)
	at org.keycloak.forms.login.freemarker.FreeMarkerLoginFormsProvider.createCommonAttributes(FreeMarkerLoginFormsProvider.java:485)
```

### How to Reproduce?

n/a

**Comment by trixpan** [comment]
@pedroigor FYI as the original commit was yours. :-)

**Comment by trixpan** [comment]
The fix is for tests to have

```
    @BeforeEach
    void setUp() throws Exception {
        Profile profile = Profile.defaults();
        ...
        }
``` 

However I couldn't find a reference to that in Upgrade or Changelogs

**Comment by pedroigor** [comment]
Hey, @trixpan. We don't support anything other than running our test suite which is mostly integration tests. In this case, it is expected that whenever you are testing Keycloak you must have the `Profile` object initialized as it is used in several places (not only for organizations) to check the state of features when at runtime.

I'm closing it as we won't provide any fix for it. What you did sounds reasonable to me due to the reasons above.

**Comment by AlexonIvy** [comment]
All I can say is that when I want to do my first experiences with Keycloak, I do not get any admin user. And when I try to create it manually, I get this error. Using docker compose on Debian in WSL 2.

```shell
bash-5.1$ /opt/keycloak/bin/kc.sh bootstrap-admin --user admin --password xxx --verbose
Changes detected in configuration. Updating the server image.
ERROR: Failed to run 'build' command.
Error details:
java.lang.NullPointerException: Cannot read field "features" because the return value of "org.keycloak.common.Profile.getInstance()" is null
``` 

This is my docker-compose.yml:

```dockerfile
services:
  keycloak:
    image: quay.io/keycloak/keycloak
    container_name: keycloak1
    restart: always
    environment:
      - KC_DB=postgres
      - KC_DB_URL=jdbc:postgresql://mydb.com:5432/keycloak1
      - KC_DB_USERNAME=keycloak1
      - KC_DB_PASSWORD=XXX
      - KC_HOSTNAME=myhostcom
      - KC_HTTP_ENABLED=true
      - KC_LOG_LEVEL=info
      - KC_BOOTSTRAP_ADMIN_USERNAME=admin
      - KC_BOOTSTRAP_ADMIN_PASSWORD=YYY
      - KEYCLOAK_ADMIN=admin
      - KEYCLOAK_ADMIN_PASSWORD=YYY
   ports:
    - "8080:8080"
    command: ["start-dev"]
``` 

Postgres was initialised like this:
```
CREATE ROLE keycloak1 WITH LOGIN PASSWORD 'XXX';
CREATE DATABASE keycloak1 WITH OWNER keycloak1;
GRANT ALL PRIVILEGES ON DATABASE keycloak1 TO keycloak1;
GRANT USAGE ON SCHEMA public TO keycloak1;
GRANT CREATE ON SCHEMA public TO keycloak1;
``` 

Sorry if this is the wrong place, just wanted to report that whatever this is, it actually happens.

**Comment by ericmorand** [comment]
Yes, this is happening. We can't bootstrap the admin, using the official Keycloak command:

```sh
$ bin/kc.sh bootstrap-admin
Changes detected in configuration. Updating the server image.
ERROR: Failed to run 'build' command.
Error details:
java.lang.NullPointerException: Cannot read field "features" because the return value of "org.keycloak.common.Profile.getInstance()" is null
```

@pedroigor , please either reopen or explain what we are doing wrong.

**Comment by shawkins** [comment]
@ericmorand please create a new issue. As far as I can see that has been resolved for 26.4 and main, but we may want to consider a backport for 26.2.

**Comment by ericmorand** [comment]
@shawkins , actually I was using 26.3.3, and upgrading to 26.4 fixed the issue entirely. Thanks a lot for the help.
